ui/src/common/engine.ts

```diff
@@ -412,14 +413,39 @@ export abstract class Engine {
     return result.firstRow({cnt: NUM}).cnt;
   }
 
+  /**
+   * Obtain a range of timestamps to which the entire timeline is restricted.
+   */
+  get timelineConstraint(): Span<TPTime, TPDuration> | undefined {
+    return this._timelineConstraint;
+  }
+
+  /**
+   * Set or clear the range of timestamps to which the entire timeline is restricted.
+   */
+  set timelineConstraint(timeSpan: Span<TPTime, TPDuration> | undefined) {
+    this._timelineConstraint = timeSpan;
+  }
```

Comment:
I'm still learning the code base, so please excuse me my question. Is there any reason why you are using here a `get` and `set` instead of just creating a public variable? Semantically, for the outside, there is currently no difference? 

Or is it just a guideline used in this repository?

Reply:
Following the pattern established by the Perfetto team..